moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp

```diff
@@ -76,17 +77,61 @@ moveit::core::RobotStatePtr CurrentStateMonitor::getCurrentState() const
   return moveit::core::RobotStatePtr(result);
 }
 
-ros::Time CurrentStateMonitor::getCurrentStateTime() const
+ros::Time CurrentStateMonitor::getCurrentStateTimeHelper(const std::string& group) const
+{
+  const std::vector<const moveit::core::JointModel*>* active_joints = &robot_model_->getActiveJointModels();
+  if (!group.empty())
+  {
+    const moveit::core::JointModelGroup* jmg = robot_model_->getJointModelGroup(group);
+    if (jmg)
+    {
+      active_joints = &jmg->getActiveJointModels();
+    }
+    else
+    {
+      ROS_ERROR_STREAM_NAMED(LOGNAME, "There is no group with the name " << std::quoted(group) << "!");
+      return ros::Time(0.0);
+    }
+  }
+  std::optional<ros::Time> oldest_state_time = std::nullopt;
+  for (const moveit::core::JointModel* joint : *active_joints)
+  {
+    std::map<const moveit::core::JointModel*, ros::Time>::const_iterator it = joint_time_.find(joint);
+    if (it == joint_time_.end())
+    {
+      ROS_DEBUG_NAMED(LOGNAME, "Joint '%s' has never been updated", joint->getName().c_str());
+    }
+    else
+    {
+      if (oldest_state_time)
+      {
+        *oldest_state_time = std::min(*oldest_state_time, it->second);
+      }
+      else
+      {
+        oldest_state_time = it->second;
+      }
+    }
+  }
+  if (oldest_state_time)
+  {
+    return *oldest_state_time;
+  }
+  return ros::Time(0.0);
```

Comment:
You could directly return `oldest_state_time` then:
```suggestion
  return oldest_state_time;
```

Reply:
Done.